src/qibolab/instruments/emulator/pulse_simulator.py

```diff
@@ -0,0 +1,758 @@
+"""Pulse simulator module for running quantum dynamics simulation of model of
+device."""
+
+import copy
+import operator
+from typing import Dict, List, Union
+
+import numpy as np
+
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
+from qibolab.instruments.abstract import Controller
+from qibolab.instruments.emulator.engines.qutip_engine import QutipSimulator
+from qibolab.instruments.emulator.models import general_no_coupler_model
+from qibolab.platform import Coupler, Qubit
+from qibolab.pulses import PulseSequence, ReadoutPulse
+from qibolab.qubits import QubitId
+from qibolab.result import IntegratedResults, SampleResults
+from qibolab.sweeper import Parameter, Sweeper, SweeperType
+
+AVAILABLE_SWEEP_PARAMETERS = {
+    Parameter.amplitude,
+    Parameter.duration,
```

Comment:
This is actually specific to the readout pulses. Drive pulse duration sweepers works as expected. More generally I don't think there is a need to comment on this here because the sweeper function itself works properly with readout pulses, but it is 'broken' because there is no model for a readout pulse in place. So the comment perhaps should be somewhere higher up, maybe in qibolab.rst?

Reply:
I see. I already put all protocols which are currently compatible with qibocal in `qibolab.rst` so we should be fine.

Reply:
Thanks!